src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingTask.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.TestPlatform.Build.Trace;
+
+namespace Microsoft.TestPlatform.Build.Tasks;
+
+public class VSTestForwardingTask : Task, ITestTask
+{
+    private int _activeProcessId;
+
+    private const string DotnetExe = "dotnet";
+
+    [Required]
+    public ITaskItem? TestFileFullPath { get; set; }
+    public string? VSTestSetting { get; set; }
+    public ITaskItem[]? VSTestTestAdapterPath { get; set; }
+    public string? VSTestFramework { get; set; }
+    public string? VSTestPlatform { get; set; }
+    public string? VSTestTestCaseFilter { get; set; }
+    public string[]? VSTestLogger { get; set; }
+    public bool VSTestListTests { get; set; }
+    public string? VSTestDiag { get; set; }
+    public string[]? VSTestCLIRunSettings { get; set; }
+    [Required]
+    public ITaskItem? VSTestConsolePath { get; set; }
+    public ITaskItem? VSTestResultsDirectory { get; set; }
+    public string? VSTestVerbosity { get; set; }
+    public string[]? VSTestCollect { get; set; }
+    public bool VSTestBlame { get; set; }
+    public bool VSTestBlameCrash { get; set; }
+    public string? VSTestBlameCrashDumpType { get; set; }
+    public bool VSTestBlameCrashCollectAlways { get; set; }
+    public bool VSTestBlameHang { get; set; }
+    public string? VSTestBlameHangDumpType { get; set; }
+    public string? VSTestBlameHangTimeout { get; set; }
+    public ITaskItem? VSTestTraceDataCollectorDirectoryPath { get; set; }
+    public bool VSTestNoLogo { get; set; }
+    public string? VSTestArtifactsProcessingMode { get; set; }
+    public string? VSTestSessionCorrelationId { get; set; }
+
+    public override bool Execute()
+    {
+        var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
```

Comment:
Could we add:

1. An internal ctor taking `IEnvironmentVariableHelper` as argument.
2. Save this argument as a field
3. A public parameterless ctor calling the internal ctor with `new EnvironmentVariableHelper()`
4. Use the field instead of the static `Environment.GetEnvironmentVariable` here and below

This would help with testing via mocking.